### PR TITLE
Cm management point

### PR DESCRIPTION
**Pull request by jeffotterpohl**

### CODE REVIEW

**Comment by rchristman89** [comment]
```diff
diff --git a/source/DSCResources/DSC_CMManagementPoint/DSC_CMManagementPoint.psm1 b/source/DSCResources/DSC_CMManagementPoint/DSC_CMManagementPoint.psm1
index 9b90c3f..825107c 100644
--- a/source/DSCResources/DSC_CMManagementPoint/DSC_CMManagementPoint.psm1
+++ b/source/DSCResources/DSC_CMManagementPoint/DSC_CMManagementPoint.psm1
@@ -65,7 +65,7 @@ function Get-TargetResource
                                     }
                 'Username'          {
                                         $userConnection = $mpProp.Value2
-                                        if (($mpProp.Value2 -eq '') -or ($null -eq $mpProp.Value2))
+                                        if ([string]::IsNullOrEmpty($mp.Value2))
                                         {
                                             $computerAccount = $true
                                         }

```

File: source/DSCResources/DSC_CMManagementPoint/DSC_CMManagementPoint.psm1 [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+        This will set the desired state.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER SiteServerName
+        Specifies the SiteServer to install the role on.
+
+    .PARAMETER SqlServerFqdn
+        Specifies the SQL server fqdn if using a SQL replica.
```
Comment (rchristman89) [inline source/DSCResources/DSC_CMManagementPoint/DSC_CMManagementPoint.psm1:159]: FQDN

File: source/DSCResources/DSC_CMManagementPoint/DSC_CMManagementPoint.psm1 [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+
+            if (($PSBoundParameters.EnableSsl -eq $false) -or
+                ([string]::IsNullOrEmpty($PSBoundParameters.EnableSsl) -and
+                ([string]::IsNullOrEmpty($state.EnableSSL) -or $state.EnableSSL -eq $false)))
+            {
+                throw 'When CloudGateway is enabled SSL must also be enabled'
+            }
+        }
+
+        if ((($SqlServerFqdn) -and [string]::IsNullOrEmpty($DatabaseName)) -or
```
Comment (rchristman89) [inline source/DSCResources/DSC_CMManagementPoint/DSC_CMManagementPoint.psm1:284]: This:($SqlServerFqdn)
Can be $SqlServerFqdn
But I understand if you keep it the same to show the seperation.
Same with $DatabaseName

File: tests/Unit/CMManagementPoint.tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+{
+    InModuleScope $script:dscResourceName {
+        $moduleResourceName = 'ConfigMgrCBDsc - DSC_CMManagementPoint'
+
+        $getInput = @{
+            SiteCode       = 'Lab'
+            SiteServerName = 'MP.contoso.com'
+        }
+
+        $getMpReturnNoSQL = @{
```
Comment (rchristman89) [inline tests/Unit/CMManagementPoint.tests.ps1:47]: Is this a valid config for a box to not have SQL installed or not point to an SQL box?
Question is to know if I should do additonal tests.

File: tests/Unit/CMManagementPoint.tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+                    $result.UseComputerAccount    | Should -Be -ExpectedValue $true
+                    $result.SQLServerFqdn         | Should -Be -ExpectedValue ''
+                    $result.SqlServerInstanceName | Should -Be -ExpectedValue $null
+                    $result.DatabaseName          | Should -Be -ExpectedValue ''
+                    $result.Username              | Should -Be -ExpectedValue ''
+                    $result.Ensure                | Should -Be -ExpectedValue 'Present'
+                }
+
+                It 'Should return desired result when management is currently installed with local SQL' {
+                    Mock -CommandName Get-CMManagementPoint -MockWith { $getMpReturnLocalSQL  }
```
Comment (rchristman89) [inline tests/Unit/CMManagementPoint.tests.ps1:288]: extra space at the end.

File: source/DSCResources/DSC_CMManagementPoint/DSC_CMManagementPoint.psm1 (hunk 2 above)
Comment (jeffotterpohl) [reply source/DSCResources/DSC_CMManagementPoint/DSC_CMManagementPoint.psm1:293]: kept as was for separation

File: tests/Unit/CMManagementPoint.tests.ps1 (hunk 3 above)
Comment (jeffotterpohl) [reply tests/Unit/CMManagementPoint.tests.ps1:47]: SQL isn't required to specify a SQL server, connection will work, just will not actually connect to anything

**Comment by NEllis280** [comment]
Tested and everything looks good. Ryan's comment appears to be resolved.

**Pull request merged**